src/hotspot/share/gc/shared/c2/barrierSetC2.hpp

```diff
@@ -302,6 +303,12 @@ class BarrierSetC2: public CHeapObj<mtGC> {
   virtual void emit_stubs(CodeBuffer& cb) const { }
 
   static int arraycopy_payload_base_offset(bool is_array);
+
+#ifndef PRODUCT
+  virtual void dump_barrier_data(const MachNode* mach, outputStream* st) const {
+    st->print("%x", mach->barrier_data());
```

Comment:
Just wondering, do we ever have barrier data without ZGC?

Reply:
No, this is just a generic fallback implementation in case other GCs want to adopt ZGC's very late barrier expansion model. Would you prefer to remove the `st->print` statement? Either way is fine by me.

Reply:
I was wondering if we should add `Unimplemented()` instead but I don't have a strong opinion either and leave that to you.

Reply:
Thanks, I had not considered that option. If none of us have a strong opinion I will go with the current version, since it provides a minimal but still sensible default without forcing other GCs to develop a custom implementation should they adopt ZGC's barrier expansion model.